rails/date-methods.md

```diff
@@ -0,0 +1,210 @@
+## Dates in Rails
+
+Rails provides a lot of methods to play with dates. These are some of them:
+
+Today
+```ruby
+Date.today
+=> Tue, 09 Feb 2016
+```
+
+Yesterday
+```ruby
+Date.yesterday
+=> Mon, 08 Feb 2016
+```
+
+Tomorrow
+```ruby
+Date.tomorrow
+=> Wed, 10 Feb 2016
+```
+
+Create a date
+```ruby
+date = Date.new(2016, 2, 9)
+=> Tue, 09 Feb 2016
+```
+
+Beginning of the month
+```ruby
+date.beginning_of_month
+=> Mon, 01 Feb 2016
+```
+
+End of the month
+```ruby
+date.end_of_month
+=> Mon, 29 Feb 2016
+```
+
+Beginning of the year
+```ruby
+date.beginning_of_year
+=> Fri, 01 Jan 2016
+```
+
+End of the year
+```ruby
+date.end_of_year
+=> Sat, 31 Dec 2016
+```
+
+Next month
+```ruby
+date.next_month
+=> Wed, 09 Mar 2016
+```
+
+Last month
+```ruby
+date.last_month
+=> Sat, 09 Jan 2016
+```
+
+Next year
+```ruby
+date.next_year
+=> Thu, 09 Feb 2017
+```
+
+Last year
+```ruby
+date.last_year
+=> Mon, 09 Feb 2015
+```
+
+### combinations
+
+Beginning of the next month
+```ruby
+date.next_month.beginning_of_month
+=> Tue, 01 Mar 2016
+```
+
+
+### Date manipulation
+
+Adds 30 seconds
+```ruby
+Time.now + 30.seconds
+=> 2016-02-09 10:01:57 +0000
+```
+
+Adds 1 day
+```ruby
+date + 1.day
+=> Wed, 10 Feb 2016
+```
+
+Adds 1 day from a parameterised date
+```ruby
+1.day.since(date = Date.today)
```

Comment:
My bad — you don't need the `(date = Date.today)`, I was trying to give you the method signature!

- `3.days.since` will default to `since(Date.today)`
- `date = Date.new(2014,3,10) ; 5.days.since(date)` will give you 5 days since the given date